## Cleaning & Analyzing Ebay Kleinanzeigan Data

In this project, we will be taking a dataset of 50,000 points from the Ebay Kleinanzeigen classifieds section of German Ebay and cleaning and analyzing the data. We will import the dataset using Pandas to clean and organize it for analysis.

We will ultimately look at the top 6 brands and their mean price and mileage to compare them.

In [1]:
#importing libraries
import pandas as pd
import numpy as np

In [2]:
#reading the dataset into a DataFrame
autos = pd.read_csv("autos.csv", encoding='Latin-1')

In [3]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offerType    price   abtest vehicleType  yearOfRegistration  \
0      privat   Angebot   $5,000  control         bus                2004   
1      privat   Angebot   $8,500  control   limousine                1997   
2      privat   Angebot   $8,990     test   limousine                2009   
3      privat   Angebot   $4,350  control  kleinwagen                2007   
4      privat   Angebot   $1,350     test       kombi                2003   
...       ...       ...      ...      ...         ...                 ...   
49995  privat   Angebot  $24,900  control   limousine                2011   
49996  privat   Angebot   $1,980  control      cabrio                1996   
49997  privat   Angebot  $13,200     test      cabrio                2014   
49998  privat   Angebot  $22,900  control       kombi                2013   
49999  privat   Angebot   $1,250  control   limousine                1996   

         gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0        manuell      158  andere  150,000km                    3      lpg   
1      automatik      286     7er  150,000km                    6   benzin   
2        manuell      102    golf   70,000km                    7   benzin   
3      automatik       71  fortwo   70,000km                    6   benzin   
4        manuell        0   focus  150,000km                    7   benzin   
...          ...      ...     ...        ...                  ...      ...   
49995  automatik      239      q5  100,000km                    1   diesel   
49996    manuell       75   astra  150,000km                    5   benzin   
49997  automatik       69     500    5,000km                   11   benzin   
49998    manuell      150      a3   40,000km                   11   diesel   
49999    manuell      101  vectra  150,000km                    1   benzin   

            brand notRepairedDamage          dateCreated  nrOfPictures  \
0         peugeot              nein  2016-03-26 00:00:00             0   
1             bmw              nein  2016-04-04 00:00:00             0   
2      volkswagen              nein  2016-03-26 00:00:00             0   
3           smart              nein  2016-03-12 00:00:00             0   
4            ford              nein  2016-04-01 00:00:00             0   
...           ...               ...                  ...           ...   
49995        audi              nein  2016-03-27 00:00:00             0   
49996        opel              nein  2016-03-28 00:00:00             0   
49997        fiat              nein  2016-04-02 00:00:00             0   
49998        audi              nein  2016-03-08 00:00:00             0   
49999        opel              nein  2016-03-13 00:00:00             0   

       postalCode             lastSeen  
0           79588  2016-04-06 06:45:54  
1           71034  2016-04-06 14:45:08  
2           35394  2016-04-06 20:15:37  
3           33729  2016-03-15 03:16:28  
4           39218  2016-04-01 14:38:50  
...           ...                  ...  
4999

In [4]:
#exploring the DF
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

From the data's info, we can see that there are supposed to be 50,000 rows and 20 columns. In those 20 columns, we can see that 5 have some missing info (null values) because their rows do not equal 50k. We can also see that a couple of the columns that should be int64 dtype have a dtype of object. When looking at the head of the info, we can see why these columns are objects instead of integers and how to convert them. The price column is an object due to it having the '$' symbol in front of each price. The odometer has 'km' after each integer. To convert them, we will need to remove those characters and change the column name to include them. From the head, we can also see there are some null values, but in this German data set they are listed as "nein", the German word for "none." We can also see in the info that the column names are in CamelCase instead of snakecase, meaning we would need to convert them in order to replace whitespace with underscores.

## Exploring and cleaning column names

In [5]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [6]:
autos.columns = ['date_crawled', 'name', 'seller',
            'offer_type', 'price', 'ab_test',
            'vehicle_type', 'registration_year', 'gearbox',
            'power_PS', 'model', 'odometer', 
            'registration_month', 'fuel_type', 'brand',
            'unrepaired_damage', 'ad_created', 'nr_of_pictures',
            'postal_code', 'last_seen',]

In [7]:
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gearbox', 'power_PS', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code',
       'last_seen'],
      dtype='object')

In [8]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price  ab_test vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_PS   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  nr_of_pictures  \
0     peugeot              nein  2016-03-26 00:00:00               0   
1         bmw              nein  2016-04-04 00:00:00               0   
2  volkswagen              nein  2016-03-26 00:00:00               0   
3       smart              nein  2016-03-12 00:00:00               0   
4        ford              nein  2016-04-01 00:00:00               0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

Above, I changed the column names from CamelCase to snakecase in order to be able to change whitespaces to underscores. Snakecase is python's more preferred method of typing column names.

## Exploring and cleaning columns

In [9]:
autos.describe(include='all')

date_crawled         name  seller offer_type  price ab_test  \
count                 50000        50000   50000      50000  50000   50000   
unique                48213        38754       2          2   2357       2   
top     2016-04-02 15:49:30  Ford_Fiesta  privat    Angebot     $0    test   
freq                      3           78   49999      49999   1421   25756   
mean                    NaN          NaN     NaN        NaN    NaN     NaN   
std                     NaN          NaN     NaN        NaN    NaN     NaN   
min                     NaN          NaN     NaN        NaN    NaN     NaN   
25%                     NaN          NaN     NaN        NaN    NaN     NaN   
50%                     NaN          NaN     NaN        NaN    NaN     NaN   
75%                     NaN          NaN     NaN        NaN    NaN     NaN   
max                     NaN          NaN     NaN        NaN    NaN     NaN   

       vehicle_type  registration_year  gearbox      power_PS  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  nr_of_pictures   postal_code            last_seen  
count                 50000         50000.0  50000.000000                50000  
unique                   76             NaN           NaN                39481  
top     2016-04-03 00:00:00             NaN           NaN  2016-04-07 06:17:27  
freq                   1946             NaN           NaN                    8  
mean                    NaN             0.0  50813.627300                  NaN  
std                     NaN             0.0  25779.747957                  NaN  
min                     NaN             0.0   1067.000000                  NaN  
25%                     NaN             0.0  30451.000000                  NaN  
50%                     NaN             0.0  49577.000000                  NaN  
75%                     NaN             0.0  71540.000000                  NaN  
max                     NaN             0.0  99998.000000                  NaN

The columns 'seller,' 'offer_type', 'ab_test,' and 'gearbox' all have only 2 values, which means they could probably be dropped. 'registration_year,' 'power_PS,' 'registration_month,' 'nr_of_pictures,' and 'postal_code' all have NaN values in the 'unique' row, as well as most other rows, meaning they need further investigation. The columns that have numeric values stored as test which need cleaning are 'price' and 'odometer.'

In [10]:
autos['price'].describe()

count     50000
unique     2357
top          $0
freq       1421
Name: price, dtype: object

In [11]:
#cleaning 'price' column and converting to integer
autos['price'] = autos['price'].str.replace('$', '').str.replace(',', '').astype(int)

In [12]:
#cleaning 'odometer' column and converting to integer
autos['odometer'] = autos['odometer'].str.replace('km', '').str.replace(',', '').astype(int)

In [13]:
#renaming 'odometer' to 'odometer_km' to clarify it's in kilometers
autos.rename(columns={'odometer': 'odometer_km'}, inplace=True)

In [14]:
#exploring unique values for 'price'
autos['price'].unique().shape

(2357,)

In [15]:
#checking cleaned column 'price' stats
autos['price'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [16]:
#exploring unique value counts of 'price' column to see outliers and most common values
autos['price'].value_counts().sort_index()

0           1421
1            156
2              3
3              1
5              2
            ... 
10000000       1
11111111       2
12345678       3
27322222       1
99999999       1
Name: price, Length: 2357, dtype: int64

In [17]:
#looking at top of price list in ascending order to remove outliers
autos['price'].value_counts().head(10)

0       1421
500      781
1500     734
2500     643
1000     639
1200     639
600      531
800      498
3500     498
2000     460
Name: price, dtype: int64

In [18]:
#looking at bottom of price list in descending order to remove outliers
autos['price'].value_counts().sort_index(ascending=False).head(10)

99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
Name: price, dtype: int64

In [19]:
#updating list to remove some outliers
autos = autos.loc[autos['price'].between(1,900000)]

In [20]:
#exploring cleaned price list
autos['price'].describe()

count     48565.000000
mean       5888.935591
std        9059.854754
min           1.000000
25%        1200.000000
50%        3000.000000
75%        7490.000000
max      350000.000000
Name: price, dtype: float64

In [21]:
autos['price'].value_counts().sort_index(ascending=True).head()

1    156
2      3
3      1
5      2
8      1
Name: price, dtype: int64

The 'price' column seemed to have some outliers since the highest frequency row was '0' which would mean that either the person didn't enter a price in that field or they're giving their car away for free. Some had prices of $10, some $100 which are highly unlikely to be real, and some had over 1,000,000 which also seems unlikely to be real. To remove these outliers, I reran the column with only the prices between 100 and 900,000.

After removing the outliers in the 'price' column, we can see that the calculations of mean, standard deviation, minimum, maximum, and the percentages all make much more sense and seem to be normal for what you would see on a used car sales forum.

In [22]:
#exploring unique values for 'odometer_km'
autos['odometer_km'].unique().shape

(13,)

In [23]:
#looking at 'odometer_km' column stats
autos['odometer_km'].describe()

count     48565.000000
mean     125770.101925
std       39788.636804
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [24]:
#exploring value counts for 'odometer_km'
autos['odometer_km'].value_counts()

150000    31414
125000     5057
100000     2115
90000      1734
80000      1415
70000      1217
60000      1155
50000      1012
5000        836
40000       815
30000       780
20000       762
10000       253
Name: odometer_km, dtype: int64

For the 'odometer_km' column, there doesn't seem to be any outliers, so I left the column as-is. We can see from the values that the most frequent odometer reading is 150,000 31,228 entries. We can also see that the majority of cars listed for sale have more than 100,000 km on the odometer, and that any cars with 40,000 or less km have less than 1000 listings respectively.

## Exploring auto listing and view data

In [25]:
date_crawl = (autos['date_crawled'].str[:10])
date_crawl.value_counts(normalize=True, dropna=False).sort_index()

2016-03-05    0.025327
2016-03-06    0.014043
2016-03-07    0.036014
2016-03-08    0.033296
2016-03-09    0.033090
2016-03-10    0.032184
2016-03-11    0.032575
2016-03-12    0.036920
2016-03-13    0.015670
2016-03-14    0.036549
2016-03-15    0.034284
2016-03-16    0.029610
2016-03-17    0.031628
2016-03-18    0.012911
2016-03-19    0.034778
2016-03-20    0.037887
2016-03-21    0.037373
2016-03-22    0.032987
2016-03-23    0.032225
2016-03-24    0.029342
2016-03-25    0.031607
2016-03-26    0.032204
2016-03-27    0.031092
2016-03-28    0.034860
2016-03-29    0.034099
2016-03-30    0.033687
2016-03-31    0.031834
2016-04-01    0.033687
2016-04-02    0.035478
2016-04-03    0.038608
2016-04-04    0.036487
2016-04-05    0.013096
2016-04-06    0.003171
2016-04-07    0.001400
Name: date_crawled, dtype: float64

The 'date_crawled' column shows the percent of listings that were crawled by the browser on each of the listed dates. We can see it's mostly similar but tapers off towards the last few dates as those are the most recent and would have the least amount of listings crawled.

In [26]:
ad_create = (autos['ad_created'].str[:10])
ad_create.value_counts(normalize=True, dropna=False).sort_index()

2015-06-11    0.000021
2015-08-10    0.000021
2015-09-09    0.000021
2015-11-10    0.000021
2015-12-05    0.000021
                ...   
2016-04-03    0.038855
2016-04-04    0.036858
2016-04-05    0.011819
2016-04-06    0.003253
2016-04-07    0.001256
Name: ad_created, Length: 76, dtype: float64

From the 'ad_created' column we can see there were far less listings in the beginning of the data (June 2015 - February 2016) than in the latter data in March and April. If this data is what is currently listed, this makes sense as we would expect that most of the earlier listings have been sold/taken down already versus the more current listings. This can also tell us that most listings usually will be taken down in only 2 months, but that there will be some stragglers each month.

In [27]:
last_see = (autos['last_seen'].str[:10])
last_see.value_counts(normalize=True, dropna=False).sort_index()

2016-03-05    0.001071
2016-03-06    0.004324
2016-03-07    0.005395
2016-03-08    0.007413
2016-03-09    0.009595
2016-03-10    0.010666
2016-03-11    0.012375
2016-03-12    0.023783
2016-03-13    0.008895
2016-03-14    0.012602
2016-03-15    0.015876
2016-03-16    0.016452
2016-03-17    0.028086
2016-03-18    0.007351
2016-03-19    0.015834
2016-03-20    0.020653
2016-03-21    0.020632
2016-03-22    0.021373
2016-03-23    0.018532
2016-03-24    0.019767
2016-03-25    0.019211
2016-03-26    0.016802
2016-03-27    0.015649
2016-03-28    0.020859
2016-03-29    0.022341
2016-03-30    0.024771
2016-03-31    0.023783
2016-04-01    0.022794
2016-04-02    0.024915
2016-04-03    0.025203
2016-04-04    0.024483
2016-04-05    0.124761
2016-04-06    0.221806
2016-04-07    0.131947
Name: last_seen, dtype: float64

By the title of the column, 'last_seen', as well as the data we can assume that this column refers to the last time the listing was seen by a consumer/user of the site. As we would expect, the values ascend as the dates become recent. This would make sense because most listings that are still available on the site should have been seen recently and most of the ones that haven't been seen for months should be taken down already, so again we seem to have some stragglers that need to be taken down from the site.

## Cleaning 'registration_year'

In [28]:
autos['registration_year'].describe()

count    48565.000000
mean      2004.755421
std         88.643887
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

There is clearly some incorrect information in this column since the minimum registration year can't possibly be 1000 and the maximum can't possibly be 9999. This means we will need to clean this column to remove these outliers and analyze these values correctly.

In [29]:
autos = autos.loc[autos['registration_year'].between(1900,2016)]
autos['registration_year'].value_counts(normalize=True).sort_index()

1910    0.000107
1927    0.000021
1929    0.000021
1931    0.000021
1934    0.000043
          ...   
2012    0.028063
2013    0.017202
2014    0.014203
2015    0.008397
2016    0.026135
Name: registration_year, Length: 78, dtype: float64

By cleaning the data and only looking at the registration years 1900-2016, we can see that the years 1910-2016 seem to be the most accurate.

## Exploring and aggregating 'brand' column

In [30]:
autos['brand'].value_counts(normalize=True)

volkswagen        0.211264
bmw               0.110045
opel              0.107581
mercedes_benz     0.096463
audi              0.086566
ford              0.069900
renault           0.047150
peugeot           0.029841
fiat              0.025642
seat              0.018273
skoda             0.016409
nissan            0.015274
mazda             0.015188
smart             0.014160
citroen           0.014010
toyota            0.012703
hyundai           0.010025
sonstige_autos    0.009811
volvo             0.009147
mini              0.008762
mitsubishi        0.008226
honda             0.007840
kia               0.007069
alfa_romeo        0.006641
porsche           0.006127
suzuki            0.005934
chevrolet         0.005698
chrysler          0.003513
dacia             0.002635
daihatsu          0.002506
jeep              0.002271
subaru            0.002142
land_rover        0.002099
saab              0.001649
jaguar            0.001564
daewoo            0.001500
trabant           0.001392
r

After looking at the 'brand' column values, I've decided to aggregate the data by any brand with over 5% of listings. This gives us the top 6 brands which are: Volkswagon, BMW, Opel, Mercedes Benz, Audi, and Ford.

In [31]:
brands = autos['brand'].value_counts(normalize=True)
#creating the top 6 list of brands by filtering only the brands = or above 0.05
top_6_brands = brands[brands >= .05].index                     
top_6_brands

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford'], dtype='object')

In [32]:
#aggregating mean prices by top 6 brands
brand_mean_prices = {}
for row in top_6_brands:
    selected_rows = autos[autos['brand']==row]
    mean_price = int(selected_rows['price'].mean())
    brand_mean_prices[row] = mean_price
#creating series for top 6 brands and mean prices
bmp_series = pd.Series(brand_mean_prices)
bmp_series

volkswagen       5402
bmw              8332
opel             2975
mercedes_benz    8628
audi             9336
ford             3749
dtype: int64

From the results of the aggregate brand and mean price data, we can see that Opel brand has the lowest mean value, and Audi has the highest with Mercedes and BMW following closely behind. Volkswagon and Ford have mid-range mean values. 

In [33]:
#aggregating mean mileage by top 6 brands
brand_mean_mileage = {}
for row in top_6_brands:
    selected_rows = autos[autos['brand']==row]
    mean_mileage = int(selected_rows['odometer_km'].mean())
    brand_mean_mileage[row] = mean_mileage
#creating series for top 6 brands and mean mileage
bmm_series = pd.Series(brand_mean_mileage)
bmm_series

volkswagen       128707
bmw              132572
opel             129310
mercedes_benz    130788
audi             129157
ford             124266
dtype: int64

In [34]:
#creating new DF of top 6 brands with mean price and mean mileage
autos = pd.DataFrame({'mean_price': bmp_series, 'mean_mileage': bmm_series})
autos.sort_values('mean_price')

mean_price  mean_mileage
opel                 2975        129310
ford                 3749        124266
volkswagen           5402        128707
bmw                  8332        132572
mercedes_benz        8628        130788
audi                 9336        129157

## Analysis & Conclusion

By comparing the mean mileage and the mean price of the top 6 brands, we can see that the mileage does not seem to effect the prices of these brands. The higher priced brands actually have a higher mean mileage than nearly all the lower priced brands. This implies that these brands tend to be higher priced in general, even when they have the same or higher mileage than one of the brands with the lower price means. This makes sense when you consider Audi, Mercedes, and BMW are luxury car brands and they're initial prices are typically much higher than the other brands.